storage/replica.go

```diff
@@ -191,43 +192,56 @@ func NewReplica(desc *roachpb.RangeDescriptor, store *Store) (*Replica, error) {
 		tsCache:     NewTimestampCache(store.Clock()),
 		sequence:    NewSequenceCache(desc.RangeID),
 		pendingCmds: map[cmdIDKey]*pendingCmd{},
+		RangeID:     desc.RangeID,
 	}
-	r.setDescWithoutProcessUpdate(desc)
 
-	lastIndex, err := r.loadLastIndex()
-	if err != nil {
+	if err := r.newReplicaInner(desc); err != nil {
 		return nil, err
 	}
+
+	if r.ContainsKey(keys.SystemDBSpan.Key) {
+		r.maybeGossipSystemConfig()
```

Comment:
Oh, I see. There was a `maybeGossipSystemConfigLocked`, but you deleted it. Carry on.